crates/rust-project-goals-cli/src/cfp.rs

```diff
@@ -0,0 +1,533 @@
+        assert_eq!(result, "# 2026H1 Goals\n\nThis is for 2026 and H1.");
+        
+        // Test note removal
+        let content_with_notes = "# YYYYHN Goals\n\n> **NOTE:** This is a note that should be removed.\n> More note content.\n\nThis should stay.";
+        let result = process_template_content(content_with_notes, "2026H1", "2026h1");
+        assert_eq!(result, "# 2026H1 Goals\n\nThis should stay.");
+        
+        // Test that other blockquotes are preserved
+        let content_with_blockquote = "# YYYYHN Goals\n\n> This is a regular blockquote that should stay.\n\n> **NOTE:** This is a note that should be removed.\n> More note content.\n\nThis should stay.";
+        let result = process_template_content(content_with_blockquote, "2026H1", "2026h1");
+        assert_eq!(result, "# 2026H1 Goals\n\n> This is a regular blockquote that should stay.\n\nThis should stay.");
+    }
+    
+    #[test]
+    fn test_process_summary_content_no_existing_section() {
+        // Test adding a new section when no timeframe sections exist
+        let content = "# Summary\n\n[Introduction](./README.md)\n";
+        let result = process_summary_content(content, "2026h1", "2026h1");
+        
+        assert!(result.contains("# ⏳ 2026H1 goal process"));
+        assert!(result.contains("- [Overview](./2026h1/README.md)"));
+        assert!(result.contains("- [Proposed goals](./2026h1/goals.md)"));
+        assert!(result.contains("- [Goals not accepted](./2026h1/not_accepted.md)"));
+    }
+    
+    #[test]
+    fn test_process_summary_content_with_placeholder() {
+        // Test updating a section that has placeholder content
+        let content = "# Summary\n\n[Introduction](./README.md)\n\n# ⏳ 2026H1 goal process\n\n- [Not yet started]()\n";
+        let result = process_summary_content(content, "2026h1", "2026h1");
+        
+        assert!(result.contains("# ⏳ 2026H1 goal process"));
+        assert!(result.contains("- [Overview](./2026h1/README.md)"));
+        assert!(!result.contains("- [Not yet started]()"));
+    }
+    
+    #[test]
+    fn test_process_summary_content_with_existing_content() {
+        // Test that existing non-placeholder content is not modified
+        let content = "# Summary\n\n[Introduction](./README.md)\n\n# ⏳ 2026H1 goal process\n\n- [Already populated](./2026h1/README.md)\n";
+        let result = process_summary_content(content, "2026h1", "2026h1");
+        
+        // Should not change existing non-placeholder content
+        assert_eq!(result, content);
+    }
+    
+    #[test]
+    fn test_process_summary_content_with_other_timeframes() {
+        // Test adding a new section when other timeframe sections exist
+        let content = "# Summary\n\n[Introduction](./README.md)\n\n# ⏳ 2025H1 goal process\n\n- [Overview](./2025h1/README.md)\n";
+        let result = process_summary_content(content, "2026h1", "2026h1");
+        
+        assert!(result.contains("# ⏳ 2025H1 goal process"));
```

Comment:
No big deal but I was slightly confused by this, and incorrectly thought it updated the existing section. Maybe adding an assertion that it does not is enough.

```rust
assert!(result.contains("- [Overview](./2025h1/README.md)"));
```